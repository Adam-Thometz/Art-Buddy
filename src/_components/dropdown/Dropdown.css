.Dropdown {
  position: relative;
  display: flex;
  justify-content: center;
  width: 100%;
  font-size: 1.5rem;
}

.Dropdown-label {
  border-radius: 25px;
  width: 20vw;
  display: flex;
  justify-content: space-between;
  background-color: white;
  padding: 10px;
  border: 8px solid;
}

.Dropdown-label img {
  margin: auto 0;
  height: 20px;
  width: 30px;
}

.Dropdown-options {
  display: none;
  z-index: 25;
  background-color: white;
  padding: 10px 20px;
  height: 20vh;
  overflow: scroll;
  width: 19.75vw;
  border: 1px solid #D4D2D5;
  border-top: none;
  border-radius: 0 0 25px 25px;
  font-size: 1rem;
}

.Dropdown-options {
  position: absolute;
  top: 47px;
  display: block;
}

.Dropdown-option {
  font-size: 1.5rem;
  border-bottom: 1px solid #D4D2D5;
  cursor: pointer;
}

.Dropdown-option:hover {
  background-color: pink;
}