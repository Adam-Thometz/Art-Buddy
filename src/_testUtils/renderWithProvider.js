import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;