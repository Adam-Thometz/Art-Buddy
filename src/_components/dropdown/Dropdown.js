import React, { useState } from 'react';

import './Dropdown.css';

import downArrowIcon from './down-arrow.png';

const Dropdown = ({ id = null, labelText, onClick, options, borderColor }) => {
  const [showOptions, setShowOptions] = useState(false);
  const [displayLabelText, setDisplayLabelText] = useState(labelText);

  const handleShowOptions = () => setShowOptions(show => !show);
  
  const handleChangeLabel = e => {
    setDisplayLabelText(e.target.textContent);
    setShowOptions(show => !show);
  };

  const optionsMenu = (
    <div
      className='Dropdown-options'
      style={{ zIndex: showOptions ? 6 : 4 }}
      onClick={onClick}
      id={id !== null ? id : null}
    >
      {Object.keys(options).map((value) => (
        <p key={value} className='Dropdown-option' onClick={handleChangeLabel} id={value}>{options[value]}</p>
      ))}
    </div>
  );

  const labelStyle = {
    zIndex: showOptions ? 7 : 5,
    borderColor
  };

  const downArrow = <img src={downArrowIcon} alt='' />;
  
  return (
    <div className="Dropdown">
      <span className='Dropdown-label' onClick={handleShowOptions} style={labelStyle}>{displayLabelText} {downArrow}</span>
      {showOptions ? optionsMenu : null}
    </div>
  );
};

export default Dropdown;