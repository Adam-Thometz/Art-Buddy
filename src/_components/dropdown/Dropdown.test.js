import React from "react";

import renderWithProvider from "../../_testUtils/renderWithProvider";

import Dropdown from "./Dropdown";

const testOptions = {
  test: 'test option',
  test2: 'another test option'
}

describe('Dropdown component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Dropdown options={testOptions} />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithProvider(<Dropdown options={testOptions} />);
    expect(asFragment()).toMatchSnapshot();
  })
})