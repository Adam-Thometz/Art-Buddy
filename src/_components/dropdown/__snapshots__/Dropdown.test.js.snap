// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dropdown"
  >
    <span
      class="Dropdown-label"
      style="z-index: 5;"
    >
       
      <img
        alt=""
        src="file"
      />
    </span>
  </div>
</DocumentFragment>
`;
